ir/state.cpp

```diff
@@ -184,7 +293,9 @@ const OrExpr* State::jumpCondFrom(const BasicBlock &bb) const {
   return I == pres.end() ? nullptr : &I->second.domain.path;
 }
 
-bool State::isUndef(const expr &e) const {
+bool State::isUndef(const expr &e, const Value *used_by) const {
+  if (used_by)
+    return at(*used_by).second.count(e) != 0;
```

Comment:
does this give any advantage? I would be surprised as undef vars don't really shrink as execution goes.
If there's no advantage, just remove this stuff as it's unneeded complexity.

Reply:
It was related with correctness: this change makes `tests/alive-tv/undef/br-undef3.srctgt.ll` success.

Reply:
ah, yes, freeze shrink undef. good.